crates/core/src/timeline/lsm_tree.rs

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use crate::error::CoreError;
+use crate::storage::Storage;
+use crate::Result;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+pub const LSM_TIMELINE_DIR: &str = ".hoodie/timeline";
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TimelineManifest {
+    pub version: i64,
+    pub entries: Vec<ManifestEntry>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ManifestEntry {
+    pub file_name: String,
+    pub min_instant: String,
+    pub max_instant: String,
+    pub level: i32,
+    pub file_size: i64,
+}
+
+pub struct LSMTree {
+    storage: Arc<Storage>,
```

Comment:
need to take any hudi configs for LSM tree specific settings?

Reply:
We'll pass configs into the LSM reader when adding archived reads to honor timeline-specific settings.